#
# Submitted-By: Sascha Hauer, 2006-03-24
# Committed-By: Sascha Hauer
#
# Error:
#
# Xorg tries to set the timing values in the framebuffer device.
# Add an option to tell X to leave the timing values untouched
#
# State:
# 
# <upstream state of this problem>
#
Index: xf86-video-fbdev-X11R7.0-0.1.0.5/src/fbdev.c
===================================================================
--- xf86-video-fbdev-X11R7.0-0.1.0.5.orig/src/fbdev.c
+++ xf86-video-fbdev-X11R7.0-0.1.0.5/src/fbdev.c
@@ -114,13 +114,15 @@ static SymTabRec FBDevChipsets[] = {
 typedef enum {
 	OPTION_SHADOW_FB,
 	OPTION_ROTATE,
-	OPTION_FBDEV
+	OPTION_FBDEV,
+	OPTION_DONT_SET_TIMING
 } FBDevOpts;
 
 static const OptionInfoRec FBDevOptions[] = {
 	{ OPTION_SHADOW_FB,	"ShadowFB",	OPTV_BOOLEAN,	{0},	FALSE },
 	{ OPTION_ROTATE,	"Rotate",	OPTV_STRING,	{0},	FALSE },
 	{ OPTION_FBDEV,		"fbdev",	OPTV_STRING,	{0},	FALSE },
+	{ OPTION_DONT_SET_TIMING, "DontSetTiming", OPTV_BOOLEAN, {0},	FALSE },
 	{ -1,			NULL,		OPTV_NONE,	{0},	FALSE }
 };
 
@@ -510,9 +512,11 @@ FBDevPreInit(ScrnInfoPtr pScrn, int flag
 	}
 
 	/* select video modes */
-
-	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against framebuffer device...\n");
-	fbdevHWSetVideoModes(pScrn);
+	if( xf86ReturnOptValBool(fPtr->Options, OPTION_DONT_SET_TIMING, FALSE) == FALSE ) {
+		xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against framebuffer device...\n");
+		fbdevHWSetVideoModes(pScrn);
+	} else
+		xf86DrvMsg(pScrn->scrnIndex, X_INFO, "ignoring fb timing settings.");
 
 	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against monitor...\n");
 	{
